module/Finna/src/Finna/Recommend/Ontology.php

```diff
@@ -0,0 +1,512 @@
+     * browser session. A value of false indicates an unlimited number.
+     *
+     * @var int|bool
+     */
+    protected $maxTimesShownPerSession;
+
+    /**
+     * Parameter object representing user request.
+     *
+     * @var \Laminas\StdLib\Parameters
+     */
+    protected $request;
+
+    /**
+     * Current search query.
+     *
+     * @var string
+     */
+    protected $lookfor;
+
+    /**
+     * Search results object.
+     *
+     * @var \VuFind\Search\Base\Results
+     */
+    protected $results;
+
+    /**
+     * Search ID.
+     *
+     * @var int
+     */
+    protected $searchId;
+
+    /**
+     * Total count of records in the result set.
+     *
+     * @var int
+     */
+    protected $resultTotal;
+
+    /**
+     * Ontology results.
+     *
+     * @var array
+     */
+    protected $ontologyResults;
+
+    /**
+     * Total number of API calls made.
+     *
+     * @var int
+     */
+    protected $apiCallTotal = 0;
+
+    /**
+     * Total number of recommendations.
+     *
+     * @var int
+     */
+    protected $recommendationTotal = 0;
+
+    /**
+     * Total number of ontology results.
+     *
+     * @var int
+     */
+    protected $ontologyResultTotal = 0;
+
+    /**
+     * Ontology constructor.
+     *
+     * @param Finto                $finto                Finto connection class
+     * @param CookieManager        $cookieManager        Cookie manager
+     * @param Url                  $urlHelper            Url helper
+     * @param RecommendationMemory $recommendationMemory Recommendation memory
+     */
+    public function __construct(
+        Finto $finto, CookieManager $cookieManager, Url $urlHelper,
+        RecommendationMemory $recommendationMemory
+    ) {
+        $this->finto = $finto;
+        $this->cookieManager = $cookieManager;
+        $this->urlHelper = $urlHelper;
+        $this->recommendationMemory = $recommendationMemory;
+    }
+
+    /**
+     * Store the configuration of the recommendation module.
+     *
+     * @param string $settings Settings from searches.ini.
+     *
+     * @return void
+     */
+    public function setConfig($settings)
+    {
+        $this->rawParams = $settings;
+    }
+
+    /**
+     * Called at the end of the Search Params objects' initFromRequest() method.
+     * This method is responsible for setting search parameters needed by the
+     * recommendation module and for reading any existing search parameters that may
+     * be needed.
+     *
+     * @param \VuFind\Search\Base\Params $params  Search parameter object
+     * @param \Laminas\StdLib\Parameters $request Parameter object representing user
+     *                                            request.
+     *
+     * @return void
+     */
+    public function init($params, $request)
+    {
+        // Parse out parameters:
+        $settings = explode(':', $this->rawParams);
+        $this->maxApiCalls = empty($settings[0]) ? false : $settings[0];
```

Comment:
Tällä on sen verran monta parametria, että voisi olla parempi laittaa oma sektio searches.ini:iin ja lukea ne sieltä.

Reply:
Pohdin kyllä ihan samaa, mutta jos nämä menisivät vielä, ja sitten jos tulee yksikin uusi...? 😊

Reply:
Helpompi se on tehdä nyt, kun myöhemmin joutuu sitten huolehtimaan taaksepäin yhteensopivuudesta.

Reply:
Totta, laitetaan siis oma sektio! 😅

Reply:
Pitäisikö tämä tehdä joissakin suosittelijoissa käytettyyn tyyliin `Ontology:[ini section]:[ini name]` vai käyttää fiksattua sektiota `searches.ini`:ssä tai jossain muussa ini:ssä?

Reply:
Laittaisin fiksatun sektion nimen suosittelijan nimen mukaisesti, tässä esim. `[OntologyRecommend]`.